source/daplink/interface/swd_host.c

```diff
@@ -48,6 +57,16 @@
 #define SCB_AIRCR_PRIGROUP_Pos              8                                             /*!< SCB AIRCR: PRIGROUP Position */
 #define SCB_AIRCR_PRIGROUP_Msk             (7UL << SCB_AIRCR_PRIGROUP_Pos)                /*!< SCB AIRCR: PRIGROUP Mask */
 
+#ifdef TARGET_MCU_PSOC6
+#define SYS_AP (0u << 24u)
+#define CM0_AP (1u << 24u)
+#define CM4_AP (2u << 24u)
+
+uint32_t DAP_AP = SYS_AP;
```

Comment:
Please use a macro definition in the board yaml to set `DAP_AP` so there is not target-specific code in the common code.

Reply:
I have to use the variable DAP_AP, because in PSoC6 there are three of them, and the user can restrict access to any of them. The target_before_init_debug function determines which of the DAP_AP is currently available and will continue to work through it.

Reply:
code has been redesigned to use g_target_family-> apsel